src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java

```diff
@@ -506,32 +505,28 @@ private void triggerPreferredLeaderElection(List<TopicPartitionInfo> partitionIn
       LOGGER.info("{}: triggerPreferredLeaderElection - {}", this.getClass().toString(), electPreferredLeadersResult.all());
     }
 
-    private static void reassignPartitions(KafkaZkClient zkClient, Collection<Node> brokers, String topic, int partitionCount, int replicationFactor) {
-      scala.collection.mutable.ArrayBuffer<BrokerMetadata> brokersMetadata = new scala.collection.mutable.ArrayBuffer<>(brokers.size());
+    private static void reassignPartitions(AdminClient adminClient, Collection<Node> brokers, String topicName,
+        int partitionCount) {
+
+      Map<TopicPartition, Optional<NewPartitionReassignment>> reassignments = new HashMap<>();
+      List<Integer> targetReplicas = new ArrayList<>();
       for (Node broker : brokers) {
-        brokersMetadata.$plus$eq(new BrokerMetadata(broker.id(), Option$.MODULE$.apply(broker.rack())));
-      }
-      scala.collection.Map<Object, Seq<Object>> assignedReplicas =
-          AdminUtils.assignReplicasToBrokers(brokersMetadata, partitionCount, replicationFactor, 0, 0);
```

Comment:
Previously the call of AdminUtils.assignReplicasToBrokers would try to distribute the replicas among the set of brokers, more or less evenly.

But now every partition will have a replica on all the brokers?

Reply:
Let's try AdminUtils for the replica assignments to brokers.